lib/topic_view.rb

```diff
@@ -670,6 +675,24 @@ def published_page
     @topic.published_page
   end
 
+  def parse_mentions
+    @mentions = @posts
+      .pluck(:id, :raw)
+      .to_h { |p| [p[0], PostAnalyzer.new(p[1], nil).raw_mentions] }
+      .filter { |_, v| !v.empty? }
```

Comment:
Also if I read this correctly, that will do one more query for each post with `lookup_mentions` is that something having a dedicated table would aim to solve?

Reply:
> Any specific reason we dont rely on the fact that each post has a `post_analyzer` property?

No specific reasons, I just didn't know that property exists.

> Also if I read this correctly, that will do one more query for each post with `lookup_mentions`

I've just checked and yes it does a query per post, I haven't noticed that. The PostAnalyzer's behavior seems a bit surprising to me, I would expect it deals only with data passed to it without making additional requests to the database.

> is that something having a dedicated table would aim to solve?

No, it wasn't the reason, since I didn't notice those additional requests. It'll just make everything much simpler. Now I am loading mentions in two steps:
```ruby
    parse_mentions
    load_mentioned_users
```

and I have to be careful to not introduce N + 1. With a dedicated table I'd just preload mentions and navigate in post models. Also, handling of this case will be trivial with a dedicated table:

> Note that status appears on all mentions on all posts in a topic except of the case when you just posted a new post, and it appeared on the bottom of the topic. On such posts status won't be shown immediately for now (you'll need to reload page to see the status).

Anyway, I've changed it to using `post.post_analyzer` and unfortunately those N + 1 requests are still there. (Also surprising, because I deal with posts loaded from database and those posts are already cooked, so looks like there is no need to cook them once again 🤔)

Initially, I had some simple custom code for parsing mentions instead of `PostAnalyzer.new(p[1], nil).raw_mentions]`, then I realized that we should already have something for it, and found a PostAnalyzer, didn't expect it makes additional db requests.

I am working on it now, looking for a good solution without additional requests.